pkg/util/yaml/decoder_test.go

```diff
@@ -125,6 +126,40 @@ stuff: 1
 	}
 }
 
+func TestDecodeBrokenYAML(t *testing.T) {
+	s := NewYAMLOrJSONDecoder(bytes.NewReader([]byte(`---
+	stuff:
+broken-indent: 1
+---       
+  `)), 100)
+	obj := generic{}
+	err := s.Decode(&obj)
+	if err == nil {
+		t.Fatal("expected error with json: prefix, got no error")
+	}
+	if !strings.HasPrefix(err.Error(), "yaml: line 1:") {
```

Comment:
It does not

On Fri, May 6, 2016 at 7:12 AM, Michal Fojtik <notifications@github.com>
wrote:

> In pkg/util/yaml/decoder_test.go
> <https://github.com/kubernetes/kubernetes/pull/25038#discussion_r62316695>
> :
>
> > @@ -125,6 +126,40 @@ stuff: 1
> >  	}
> >  }
> >
> > +func TestDecodeBrokenYAML(t *testing.T) {
> > +	s := NewYAMLOrJSONDecoder(bytes.NewReader([]byte(`---
> > +	stuff:
> > +broken-indent: 1
> > +---
> > +  `)), 100)
> > +	obj := generic{}
> > +	err := s.Decode(&obj)
> > +	if err == nil {
> > +		t.Fatal("expected error with json: prefix, got no error")
> > +	}
> > +	if !strings.HasPrefix(err.Error(), "yaml: line 1:") {
>
> @lavalamp <https://github.com/lavalamp> actually I think the problem was
> tabs on the second line, I fixed this case... however yaml still reports
> line '2' where it should be '3' (I guess it does not count '---' as 'line'.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25038/files/ce8ba423ddf64050c4daed6548fa8794e2a4e282#r62316695>
>